astropy/coordinates/angles.py

```diff
@@ -582,8 +582,15 @@ def _validate_angles(self, angles=None):
         # objects, for speed.
         if angles is None:
             angles = self
-        lower = u.degree.to(angles.unit, -90.0)
-        upper = u.degree.to(angles.unit, 90.0)
+
+        if angles.unit == u.deg:
```

Comment:
Comparisons are very expensive for units, which means the present code will really slow down any other units. But for predefined ones like `u.rad` and `u.deg` one can do `is` instead. Trying to simplify even more:
```
if angles.unit is u.deg:
    upper = 90
elif angles.unit is u.rad:
    upper = np.pi / 2
else:
    upper  = u.degree.to(u.unit, 90)
lower = -upper
```
(Obviously one might as well call it `limit` here and use `limit` and `-limit` below.)